src/lib/handlers/drop.ts

```diff
@@ -0,0 +1,25 @@
+import type { Handler } from './types';
+
+type Handle = ReturnType<Handler>;
+type Fn = Parameters<Handle>[0];
+
+const handler = (() => {
```

Comment:
Shouldn't we also take a max here to only drop after max is reached?

Reply:
I was thinking the same and thought this would be the initial POC, but I'm happy to add `max` to all of the task types already

Reply:
It was a simple addition so I added it to this PR